crates/uv-client/src/registry_client.rs

```diff
@@ -516,6 +517,68 @@ impl RegistryClient {
                 .into_async_read(),
         ))
     }
+
+    /// Copy authentication from one URL to another URL if applicable.
+    ///
+    /// See [`RegistryClient::should_retain_auth`] for details on when authentication is retained.
+    #[must_use]
+    fn safe_copy_auth(request_url: &Url, mut response_url: Url) -> Url {
+        if RegistryClient::should_retain_auth(request_url, &response_url) {
+            response_url
+                .set_username(request_url.username())
+                .unwrap_or_else(|_| {
+                    warn!("Failed to transfer username to response URL: {response_url}")
```

Comment:
Can we copy over the error or log it with `debug` to ease debugging when copying fails for some reason?

Reply:
The error is empty, just `()` — annoying but yep.

Reply:
Oh, that's unfortunate